src/loaddb/load_sa_loader.cpp

```diff
@@ -1417,33 +1419,159 @@ ldr_assign_class_id (DB_OBJECT *class_, int id)
 
-  sm_downcase_name (class_name, realname, SM_MAX_IDENTIFIER_LENGTH);
+  sm_user_specified_name (class_name, realname, SM_MAX_IDENTIFIER_LENGTH);
+
   ldr_Hint_class_names[0] = realname;
 
-  find =
-	  locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
+  found = locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
 				    NULL_LOCK);
-
-  if (find == LC_CLASSNAME_EXIST)
+  if (found == LC_CLASSNAME_EXIST)
     {
       class_ = db_find_class (class_name);
+
+      ldr_Hint_class_names[0] = NULL;
+
+      return class_;
+    }
+
+  /* This is the case when the loaddb utility is executed with the --no-user-specified-name option as the dba user. */
+  if (db_get_client_type() == DB_CLIENT_TYPE_ADMIN_UTILITY && prm_get_bool_value (PRM_ID_NO_USER_SPECIFIED_NAME))
+    {
+      char other_class_name[DB_MAX_IDENTIFIER_LENGTH] = { '\0' };
+
+      ldr_find_class_by_query (realname, other_class_name, DB_MAX_IDENTIFIER_LENGTH);
+      if (other_class_name[0] != '\0')
+	{
+	  ldr_Hint_class_names[0] = other_class_name;
+
+	  found = locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
+					    NULL_LOCK);
+	  if (found == LC_CLASSNAME_EXIST)
+	    {
+	      class_ = db_find_class (other_class_name);
+
+	      ldr_Hint_class_names[0] = NULL;
+
+	      return class_;
+	    }
+	}
     }
 
   ldr_Hint_class_names[0] = NULL;
 
-  return (class_);
+  return class_;
+}
+
+static int
+ldr_find_class_by_query (const char *name, char *buf, size_t buf_size)
+{
+#define QUERY_BUF_SIZE 2048
+#define CT_CLASS_NAME "_db_class"
+  DB_QUERY_RESULT *query_result = NULL;
+  DB_QUERY_ERROR query_error;
+  DB_VALUE value;
+  const char *query = NULL;
+  char query_buf[QUERY_BUF_SIZE] = { '\0' };
+  const char *dot = NULL;
+  const char *name_p = NULL;
+  char *current_user_name = NULL;
+  int error = NO_ERROR;
+
+  db_make_null (&value);
+  query_error.err_lineno = 0;
+  query_error.err_posno = 0;
+
+  if (name == NULL || name[0] == '\0' || buf == NULL || buf_size < DB_MAX_IDENTIFIER_LENGTH)
+    {
+      ERROR_SET_WARNING (error, ER_OBJ_INVALID_ARGUMENTS);
+      return error;
+    }
+
+  dot = strchr (name, '.');
+  name_p = dot ? (dot + 1) : name;
+
+  current_user_name = db_get_user_name ();
+  if (!current_user_name)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      return error;
+    }
+
+  query = "SELECT [class_full_name] FROM [%s] WHERE [class_name] = '%s' AND [owner].[name] != UPPER ('%s')";
+  assert (QUERY_BUF_SIZE > snprintf (NULL, 0, query, CT_CLASS_NAME, name_p, current_user_name));
+  snprintf (query_buf, sizeof (query_buf), query, CT_CLASS_NAME, name_p, current_user_name);
+
+  error = db_compile_and_execute_local (query_buf, &query_result, &query_error);
+  if (error < NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      goto end;
+    }
+
+  db_private_free_and_init (NULL, current_user_name);
```

Comment:
Even if an error occurs, db_private_free_and_init() must be called.
```
 db_private_free_and_init (NULL, current_user_name);

 error = db_compile_and_execute_local (query_buf, &query_result, &query_error);
  if (error < NO_ERROR)
    {
      ASSERT_ERROR ();
      goto end;
    }
```

Reply:
Okay, I'll change it.